implementation/src/main/java/io/smallrye/openapi/runtime/scanner/AnnotationScannerExtension.java

```diff
@@ -0,0 +1,55 @@
+package io.smallrye.openapi.runtime.scanner;
+
+import org.eclipse.microprofile.openapi.models.parameters.Parameter.In;
+import org.jboss.jandex.MethodInfo;
+import org.jboss.jandex.MethodParameterInfo;
+import org.jboss.jandex.Type;
+
+import io.smallrye.openapi.runtime.util.JandexUtil.JaxRsParameterInfo;
+
+/**
+ * Extension point for supporting extensions to JAX-RS.
+ */
+public class AnnotationScannerExtension {
```

Comment:
This is a class rather than an interface?

Reply:
Yes, so that extensions don't have to implement every method.

Reply:
I wonder if an interface with default methods might better capture the intent.

Reply:
Well, yes that would work too. I guess it only depends on your minimum Java version, but if it's fine by you I can change that.

Reply:
+1 to interface. AFAIK, (cc @kenfinnigan), Smallrye expects Java >=8.

Reply:
Correct, right now it's Java 8+

Reply:
Great, then I think the consensus is interface with default methods.  Otherwise I think this looks good and we just need to wait for jandex before merging.  Thanks for the contribution!